pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -1086,13 +1086,77 @@ func validateValidatingAdmissionPolicyBindingSpec(spec *admissionregistration.Va
 			allErrors = append(allErrors, field.Invalid(fldPath.Child("policyName"), spec.PolicyName, msg))
 		}
 	}
-	allErrors = append(allErrors, validateParamRef(spec.ParamRef, fldPath.Child("paramRef"))...)
+	allErrors = append(allErrors, validateBindingParamRef(spec.ParamRef, fldPath.Child("paramRef"))...)
 	allErrors = append(allErrors, validateMatchResources(spec.MatchResources, fldPath.Child("matchResouces"))...)
 	allErrors = append(allErrors, validateValidationActions(spec.ValidationActions, fldPath.Child("validationActions"))...)
 
 	return allErrors
 }
 
+func validateBindingParamRef(pr *admissionregistration.BindingParamRef, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if pr == nil {
+		return allErrors
+	}
+
+	switch pr.Kind {
+	case "":
+		allErrors = append(allErrors, field.Required(fldPath.Child("kind"), ""))
+	case admissionregistration.BindingParamKindGlobal:
+		if pr.GlobalParamRef == nil {
+			allErrors = append(allErrors, field.Required(fldPath.Child("globalParamRef"), fmt.Sprintf("required if Kind is `%s`", admissionregistration.BindingParamKindGlobal)))
+		} else {
+			allErrors = append(allErrors, validateParamRef(pr.GlobalParamRef, fldPath.Child("globalParamRef"))...)
+		}
+	case admissionregistration.BindingParamKindNamespace:
+		if pr.NamespaceParamRef == nil {
+			allErrors = append(allErrors, field.Required(fldPath.Child("namespaceParamRef"), fmt.Sprintf("required if Kind is `%s`", admissionregistration.BindingParamKindNamespace)))
+		} else {
+			allErrors = append(allErrors, validateNamespaceParamRef(pr.NamespaceParamRef, fldPath.Child("namespaceParamRef"))...)
+		}
+	default:
+		allErrors = append(allErrors, field.NotSupported(fldPath.Child("kind"), pr.Kind, []string{string(admissionregistration.BindingParamKindGlobal), string(admissionregistration.BindingParamKindNamespace)}))
+	}
+
+	return allErrors
+}
+
+func validateNamespaceParamRef(npr *admissionregistration.NamespaceParamRef, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if npr == nil {
+		return allErrors
+	}
+
+	numParams := 0
+
+	if len(npr.Name) > 0 {
+		numParams += 1
+		for _, msg := range path.ValidatePathSegmentName(npr.Name, false) {
+			allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), npr.Name, msg))
+		}
+	}
+
+	if npr.Selector != nil {
+		numParams += 1
+		labelSelectorValidationOpts := metav1validation.LabelSelectorValidationOptions{}
+		allErrors = append(allErrors, metav1validation.ValidateLabelSelector(npr.Selector, labelSelectorValidationOpts, fldPath)...)
+	}
+
+	if numParams == 0 {
+		allErrors = append(allErrors, field.Required(fldPath, "must specify either name or selector"))
```

Comment:
Based on the doc, I expected empty to mean "select all in the namespace"

Reply:
Added doc and removed error